ksqldb-functional-tests/src/test/resources/query-validation-tests/formats.json

```diff
@@ -271,6 +274,28 @@
         ]
       }
     },
+    {
+      "name": "override key format - create stream as select (new planner)",
+      "statements": [
+        "CREATE STREAM INPUT (K STRING KEY, foo INT) WITH (kafka_topic='input_topic', key_format='KAFKA', value_format='JSON');",
+        "CREATE STREAM OUTPUT WITH (key_format='JSON') AS SELECT * FROM INPUT;"
+      ],
+      "properties": {
+        "ksql.new.query.planner.enabled": "true"
+      },
+      "inputs": [
+        {"topic": "input_topic", "key": "hello", "value": {"FOO": 10}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": {"K": "hello"}, "value": {"FOO": 10}}
```

Comment:
The new planner does not unwrap the key compared to the old planner. Need to discuss this. (same for other test below).

Reply:
What is behind this change?  And is it intentional or just part of the work-in-progress?

Reply:
Both. I wrote the code and the test failed. This is the WIP part. However, we discussed this topic in the past, and I believe the new behavior is actually better. So the question is if we want to keep the new behavior, or if we need to follow up and teach the new planner to apply the old behavior.

Reply:
What have folks like @colinhicks said about this change?